addons/account/views/account_portal_templates.xml

```diff
@@ -113,7 +113,24 @@
                 <t t-call="portal.portal_record_sidebar">
                     <t t-set="classes" t-value="'col-lg-3 col-xl-4 d-print-none me-lg-auto'"/>
                     <t t-set="title">
+                        <t t-set="amount" t-value="invoice.amount_residual if invoice.amount_residual > 0 else invoice.amount_total"/>
+                        <t t-set="is_eligible_for_early_payment_discount"
+                           t-value="invoice.invoice_date and invoice._is_eligible_for_early_payment_discount(invoice.currency_id, datetime.date.today())"/>
+
+                        <div t-if="is_eligible_for_early_payment_discount" class="alert alert-info" role="alert">
+                            An early payment discount is available for this invoice!
+                        </div>
                         <h2>
+                            <t t-if="is_eligible_for_early_payment_discount">
+                                <span t-esc="amount" t-attf-class="ms-1 h5" style="text-decoration: line-through; color: grey;" t-options='{"widget": "monetary", "display_currency": invoice.currency_id}'/>
+                                <span t-esc="invoice.invoice_payment_term_id._get_amount_due_after_discount(amount, invoice.amount_tax)"
```

Comment:
You must no do any further computation on the invoice. Everything is already stored on account.move.line:
```py
    # === Early Pay fields === #
    discount_date = fields.Date(
        string='Discount Date',
        store=True,
        help='Last date at which the discounted amount must be paid in order for the Early Payment Discount to be granted'
    )
    # Discounted amount to pay when the early payment discount is applied
    discount_amount_currency = fields.Monetary(
        string='Discount amount in Currency',
        store=True,
        currency_field='currency_id',
        aggregator=None,
    )
    # Discounted balance when the early payment discount is applied
    discount_balance = fields.Monetary(
        string='Discount Balance',
        store=True,
        currency_field='company_currency_id',
    )
```
You can retrieve the payment term lines by filtering on `aml.display_type == 'payment_term'`.